src/router/routes/users.ts

```diff
@@ -303,4 +324,149 @@ export class UsersController extends ParentRouter implements IExpressController
+   * @apiUse ResponseBodyDescription
+   * @apiUse RequestOpts
+   */
+  private async getExtraCreditAssignmentsByUidHandler(req: Request, res: Response, next: NextFunction) {
+    if (!req.query) {
+      return Util.standardErrorHandler(new HttpError('Illegal request format', 400), next);
+    }
+
+    if (!req.query.uid) {
+      return Util.standardErrorHandler(new HttpError('Could not find valid uid', 400), next);
+    }
+
+    try {
+      const uid: string = req.query.uid;
+      const result = await this.extraCreditDataMapper.getByUser(uid);
+      const response = new ResponseBody(
+        'Success',
+        200,
+        result);
+      return this.sendResponse(res, response);
+    } catch (error) {
+      return Util.errorHandler500(error, next);
+    }
+  }  
+
+  /**
+   * @api {get} /users/extra-credit?type=class Get all extra credit assignments for a class
+   * @apiVersion 2.0.0
+   * @apiName Get Extra Credit Assignments By Class
+   * @apiGroup User
+   * @apiPermission UserPermission
+   *
+   * @apiParam {Integer} cid - the id associated with the class
+   * @apiUse AuthArgumentRequired
+   *
+   * @apiSuccess {ExtraCreditAssignment} The retrieved extra credit assignments
+   * @apiUse ResponseBodyDescription
+   * @apiUse RequestOpts
+   */
+  private async getExtraCreditAssignmentsByClassHandler(req: Request, res: Response, next: NextFunction) {
+    if (!req.query) {
+      return Util.standardErrorHandler(new HttpError('Illegal request format', 400), next);
+    }
+
+    if (!req.query.cid) {
+      return Util.standardErrorHandler(new HttpError('Could not find valid class id', 400), next);
+    }
+
+    try {
+      const cid: number = req.query.cid;
+      const result = await this.extraCreditDataMapper.getByClass(cid);
+      const response = new ResponseBody(
+        'Success',
+        200,
+        result);
+      return this.sendResponse(res, response);
+    } catch (error) {
+      return Util.errorHandler500(error, next);
+    }
+  }  
+  
+ /**
+   * @api {post} /users/extra-credit/delete Remove an extra credit assignment
+   * @apiVersion 2.0.0
+   * @apiName Remove Extra Credit Assignment
+   * @apiGroup User
+   * @apiPermission DirectorPermission
+   *
+   * @apiParam {String} uid - the id associated with the hacker
+   * @apiParam {String} hackathonUid - the id associated with the current hackathon
+   * @apiUse AuthArgumentRequired
+   * @apiUse IllegalArgumentError
+   * @apiUse ResponseBodyDescription
+   */
+  private async deleteExtraCreditAssignmentHandler(
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ) {
+    if (!req.body) {
+      return Util.standardErrorHandler(new HttpError('Illegal request format', 400), next);
+    }
+    if (!req.body.uid) {
+      return Util.standardErrorHandler(new HttpError('Could not find valid assignment uid', 400), next);
+    }
+    try {
+      const id: string = req.body.uid
+      const result = await this.extraCreditDataMapper.delete(id);
```

Comment:
since delete expects an object of type ExtraCreditAssignment, you should construct one here, set the uid, and then pass that as the parameter